rust/private/rustc.bzl

```diff
@@ -655,6 +655,18 @@ def establish_cc_info(ctx, crate_info, toolchain, cc_toolchain, feature_configur
             cc_toolchain = cc_toolchain,
             static_library = crate_info.output,
         )
+    elif crate_info.type in ("rlib", "lib"):
+        # bazel hard-codes a check for endswith((".a", ".pic.a",
+        # ".lib")) in create_library_to_link, so we work around that
+        # by creating a symlink to the .rlib with a .a extension.
+        dot_a = ctx.actions.declare_file(crate_info.name + ".a", sibling = crate_info.output)
+        ctx.actions.symlink(output = dot_a, target_file = crate_info.output)
+        library_to_link = cc_common.create_library_to_link(
+            actions = ctx.actions,
+            feature_configuration = feature_configuration,
+            cc_toolchain = cc_toolchain,
+            static_library = dot_a,
```

Comment:
I think rustc produces PIC archives by default, so we could set pic_static_library to dot_a as well. Please add a todo for me to fix it to handle NOPIC correctly.

Reply:
I think I did what you asked here. PTAL?

Reply:
Yes, thank you!